common/motion_utils/include/motion_utils/trajectory/trajectory.hpp

```diff
@@ -438,40 +534,49 @@ double calcSignedArcLength(const T & points, const size_t src_idx, const size_t
 }
 
 /**
- *  @brief Computes the partial sums of the elements in the sub-ranges of
- *         the range [src_idx, dst_idx) and return these sum as vector
+ * @brief Computes the partial sums of the elements in the sub-ranges of the range [src_idx,
+ * dst_idx) and return these sum as vector
+ * @param points points of trajectory, path, ...
+ * @param src_idx index of start point
+ * @param dst_idx index of end point
+ * @return partial sums container
  */
 template <class T>
 std::vector<double> calcSignedArcLengthPartialSum(
   const T & points, const size_t src_idx, const size_t dst_idx)
 {
   try {
-    validateNonEmpty(points);
   } catch (const std::exception & e) {
+    validateNonEmpty(points);
     std::cerr << e.what() << std::endl;
     return {};
   }
-
   if (src_idx + 1 > dst_idx) {
     auto copied = points;
     std::reverse(copied.begin(), copied.end());
     return calcSignedArcLengthPartialSum(points, dst_idx, src_idx);
   }
-
   std::vector<double> partial_dist;
   partial_dist.reserve(dst_idx - src_idx);
 
   double dist_sum = 0.0;
   partial_dist.push_back(dist_sum);
   for (size_t i = src_idx; i < dst_idx - 1; ++i) {
-    dist_sum += tier4_autoware_utils::calcDistance2d(points.at(i), points.at(i + 1));
     partial_dist.push_back(dist_sum);
+    dist_sum += tier4_autoware_utils::calcDistance2d(points.at(i), points.at(i + 1));
   }
```

Comment:
Is this change okay? This common function is already used in the behavior_path_planner.

Reply:
Thanks for your comments Mr. @takayuki5168 .
I am just trying to understand the change request and its context : 
- Is it only deleting the line before puch_back and making it after it ?
  - if yes. that means the first 2 slots of the vector will be the same, as we already puch_back before the loop. Is it right ?
- Is it OK to do a code change in this PR in the time it was originally open for doxygen comments ?

Reply:
Mr. @takayuki5168 : 
I have understood the problem, it was due to a merge conflict mistake in one of previous commit and I did not realize it.

Please review again let me know if you have further findings.

Thanks again